core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -2109,35 +1898,138 @@ func (o *evmTxStore) UpdateTxAttemptBroadcastBeforeBlockNum(ctx context.Context,
+	attempts = dbEthTxAttemptsToEthTxAttempts(dbTxAttempts)
+	return attempts, err
+}
+
+// FindConfirmedTxesReceipts returns all confirmed transactions with receipt block nums older than or equal to the finalized block number
+func (o *evmTxStore) FindConfirmedTxesReceipts(ctx context.Context, finalizedBlockNum int64, chainID *big.Int) (receipts []*evmtypes.Receipt, err error) {
 	var cancel context.CancelFunc
 	ctx, cancel = o.stopCh.Ctx(ctx)
 	defer cancel()
 
+	var dbReceipts []Receipt
 	// note the receipts are partially loaded for performance reason
-	query := `SELECT evm.receipts.id, evm.receipts.tx_hash, evm.receipts.block_hash, evm.receipts.block_number FROM evm.receipts
+	query := `SELECT evm.receipts.tx_hash, evm.receipts.block_hash, evm.receipts.block_number FROM evm.receipts
 		INNER JOIN evm.tx_attempts ON evm.tx_attempts.hash = evm.receipts.tx_hash
 		INNER JOIN evm.txes ON evm.txes.id = evm.tx_attempts.eth_tx_id
 		WHERE evm.txes.state = 'confirmed' AND evm.receipts.block_number <= $1 AND evm.txes.evm_chain_id = $2`
-	err = o.q.SelectContext(ctx, &receipts, query, finalizedBlockNum, chainID.String())
+	err = o.q.SelectContext(ctx, &dbReceipts, query, finalizedBlockNum, chainID.String())
+	for _, dbReceipt := range dbReceipts {
+		receipts = append(receipts, &evmtypes.Receipt{
+			TxHash:      dbReceipt.TxHash,
+			BlockHash:   dbReceipt.BlockHash,
+			BlockNumber: big.NewInt(dbReceipt.BlockNumber),
+		})
+	}
 	return receipts, err
 }
 
-// Mark transactions corresponding to receipt IDs as finalized
-func (o *evmTxStore) UpdateTxStatesToFinalizedUsingReceiptIds(ctx context.Context, receiptIDs []int64, chainId *big.Int) error {
-	if len(receiptIDs) == 0 {
+// Mark transactions corresponding to attempt hashes as finalized
+func (o *evmTxStore) UpdateTxStatesToFinalizedUsingTxHashes(ctx context.Context, txHashes []common.Hash, chainID *big.Int) error {
+	if len(txHashes) == 0 {
 		return nil
 	}
+	txHashBytea := make([][]byte, len(txHashes))
+	for i, hash := range txHashes {
+		txHashBytea[i] = hash.Bytes()
+	}
 	var cancel context.CancelFunc
 	ctx, cancel = o.stopCh.Ctx(ctx)
 	defer cancel()
 	sql := `
 UPDATE evm.txes SET state = 'finalized' WHERE evm.txes.evm_chain_id = $1 AND evm.txes.id IN (SELECT evm.txes.id FROM evm.txes
 	INNER JOIN evm.tx_attempts ON evm.tx_attempts.eth_tx_id = evm.txes.id
-	INNER JOIN evm.receipts ON evm.receipts.tx_hash = evm.tx_attempts.hash
-	WHERE evm.receipts.id = ANY($2))
+	WHERE evm.tx_attempts.hash = ANY($2))
 `
-	_, err := o.q.ExecContext(ctx, sql, chainId.String(), pq.Array(receiptIDs))
+	_, err := o.q.ExecContext(ctx, sql, chainID.String(), txHashBytea)
 	return err
 }
+
+// FindReorgOrIncludedTxs finds transactions that have either been re-org'd or included on-chain based on the mined transaction count
+// If the mined transaction count receeds, transactions could have beeen re-org'd
+// If it proceeds, transactions could have been included
```

Comment:
Can you add another comment here mentioning that this check assumes transactions are broadcasted in ascending order and not out of order?